packages/fluentui/react/src/components/Tree/Tree.tsx

```diff
@@ -191,135 +159,136 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
-      } else {
-        if (exclusive) {
-          siblings.some(sibling => {
-            const activeSiblingIdIndex = activeItemIds.indexOf(sibling['id']);
-            if (activeSiblingIdIndex > -1) {
-              activeItemIds = removeItemAtIndex(activeItemIds, activeSiblingIdIndex);
+  onFocusParent = (parent: string) => {
+    const parentRef = this.itemsRef.get(parent);
 
-              return true;
-            }
-            return false;
-          });
-        }
-
-        activeItemIds = [...activeItemIds, id];
-      }
+    if (!parentRef || !parentRef.current) {
+      return;
+    }
 
-      this.setActiveItemIds(e, activeItemIds);
+    parentRef.current.focus();
+  };
 
-      _.invoke(predefinedProps, 'onTitleClick', e, treeItemProps);
-    },
-    onFocusParent: (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
-      const { parent } = treeItemProps;
+  onTitleClick = (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+    if (!hasSubtree(treeItemProps)) {
+      return;
+    }
 
-      if (!parent) {
-        return;
+    let { activeItemIds } = this.state;
+    const { id } = treeItemProps;
+    const { exclusive, items } = this.props;
+    const siblings = getSiblings(items, id);
+
+    const activeItemIdIndex = activeItemIds.indexOf(id);
+
+    if (activeItemIdIndex > -1) {
+      activeItemIds = removeItemAtIndex(activeItemIds, activeItemIdIndex);
+    } else {
+      if (exclusive) {
+        siblings.some(sibling => {
+          const activeSiblingIdIndex = activeItemIds.indexOf(sibling['id']);
+          if (activeSiblingIdIndex > -1) {
+            activeItemIds = removeItemAtIndex(activeItemIds, activeSiblingIdIndex);
+
+            return true;
+          }
+          return false;
+        });
       }
 
-      const { itemsForRender } = this.state;
-      const parentItemForRender = itemsForRender[parent['id']];
-
-      if (!parentItemForRender || !parentItemForRender.elementRef || !parentItemForRender.elementRef.current) {
-        return;
-      }
+      activeItemIds = [...activeItemIds, id];
+    }
 
-      parentItemForRender.elementRef.current.focus();
-      _.invoke(predefinedProps, 'onFocusParent', e, treeItemProps);
-    },
-    onFocusFirstChild: (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
-      const { id } = treeItemProps;
+    this.setActiveItemIds(e, activeItemIds);
+  };
 
-      const { itemsForRender } = this.state;
-      const currentElement = itemsForRender[id].elementRef;
+  onFocusFirstChild = (itemId: string) => {
+    const currentElement = this.itemsRef.get(itemId);
 
-      if (!currentElement || !currentElement.current) {
-        return;
-      }
+    if (!currentElement || !currentElement.current) {
+      return;
+    }
 
-      const elementToBeFocused = getNextElement(this.treeRef.current, currentElement.current);
+    const elementToBeFocused = getNextElement(this.treeRef.current, currentElement.current);
```

Comment:
We know that `this.treeRef` is a ref object, so it's stable, correct? 🤔 

Can we just pass it via Context and use it inside `TreeItem`?

`TreeItem.tsx`

```diff
-  onFocusFirstChild = (itemId: string) => {
-    const currentElement = this.itemsRef.get(itemId);
-
-    if (!currentElement || !currentElement.current) {
-      return;
-    }
-
-    const elementToBeFocused = getNextElement(this.treeRef.current, currentElement.current);
-
-    if (!elementToBeFocused) {
-      return;
-    }
-
-    elementToBeFocused.focus();
-  };
```

`TreeItem.tsx`

```diff
- const { onFocusParent, onSiblingsExpand, onFocusFirstChild, onTitleClick } = React.useContext(TreeItemContext);
+ const { onFocusParent, onSiblingsExpand, onTitleClick, treeRef } = React.useContext(TreeItemContext);

//
  const handleFocusFirstChild = e => {
    _.invoke(props, 'onFocusFirstChild', e, props);
-    onFocusFirstChild(props.id);
+   const elementToBeFocused = getNextElement(treeRef.current, contentRef.current);
+
+  if (elementToBeFocused) elementToBeFocused.focus()
  };
```

Reply:
I don't see much improvement here, we are just swapping one callback with a ref. I would rather go with the callback approach since it looks more consistent, we have all 4 callbacks passed as context.